mem/mem_test.go

```diff
@@ -26,6 +26,8 @@ func TestMemRepo(t *testing.T) {
 	ctx := context.Background()
 	store := mem.New[*someEnt]()
 
+	// Can DoesItWork provide its own generic entity? Must be unique for a given
+	// backend database?
 	test.DoesItWork(t, ctx, store, func(s string) error {
 		return store.Save(ctx, &someEnt{s})
 
```

Comment:
To elaborate: Shouldn't `DoesItWork` decided what entity it wants to save to a given database?  It seems deliberate, so I expect that I have missed the point.  Why have you allowed a database implementation to specify its own save function?